project/types_yaml.go

```diff
@@ -80,7 +84,11 @@ func (s *Command) UnmarshalYAML(tag string, value interface{}) error {
 	case []interface{}:
 		parts := make([]string, len(value))
 		for k, v := range value {
-			parts[k] = v.(string)
+			if sv, ok := v.(string); ok {
+				parts[k] = sv
+			} else {
+				return fmt.Errorf("Cannot unmarshal '%v' of type %T into a string value", v, v)
+			}
```

Comment:
This block (including some of the existing lines) are repeated in quite a few places.  Could we make this a function?

```go
func ToStringSlice(value []interface{}) ([]string, error) {
    ...
}
```

Then many of these can be just:

```go
if s.parts, err = BuildParts(value); err != nil {
    return err
}
```